File: src/main/java/c6h2cl2/mod/SolidXp/Enchantment/XpBoost.java
package c6h2cl2.mod.SolidXp.Enchantment;

import c6h2cl2.mod.SolidXp.SolidXpCore;
import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.EnumEnchantmentType;
import net.minecraft.inventory.EntityEquipmentSlot;

/**
 * Created by C6H2Cl2 on 2016/06/26.
 */
public class XpBoost extends Enchantment {

    public XpBoost(EnumEnchantmentType enchantmentType) {
        super(Rarity.RARE, enchantmentType,new EntityEquipmentSlot[]{EntityEquipmentSlot.MAINHAND});
        setName("xpBoost."+enchantmentType.name());
        setRegistryName(SolidXpCore.DOMAIN,"xpBoost."+enchantmentType.name());
    }

    @Override
    public int getMaxLevel(){
        return 5;
    }

    @Override
    public int getMinEnchantability(int enchantmentLevel){
        return 15 + enchantmentLevel * 8;
    }

    @Override
    public int getMaxEnchantability(int enchantmentLevel){
        return getMinEnchantability(enchantmentLevel) + 50;
    }
}

File: src/main/java/c6h2cl2/mod/SolidXp/Enchantment/XpBoost.kt
package c6h2cl2.mod.SolidXp.Enchantment

import c6h2cl2.mod.SolidXp.SolidXpCore
import net.minecraft.enchantment.Enchantment
import net.minecraft.enchantment.EnumEnchantmentType
import net.minecraft.inventory.EntityEquipmentSlot

class XpBoost(enchantmentType: EnumEnchantmentType) : Enchantment(Enchantment.Rarity.RARE, enchantmentType, arrayOf(EntityEquipmentSlot.MAINHAND)) {

    init {
        setName("xpBoost." + enchantmentType.name)
        setRegistryName(SolidXpCore.DOMAIN, "xpBoost." + enchantmentType.name)
    }

    override fun getMaxLevel(): Int {
        return 5
    }

    override fun getMinEnchantability(enchantmentLevel: Int): Int {
        return 15 + enchantmentLevel * 8
    }

    override fun getMaxEnchantability(enchantmentLevel: Int): Int {
        return getMinEnchantability(enchantmentLevel) + 50
    }
}
